standards/application/waku-api.md

```diff
@@ -0,0 +1,327 @@
+editor: Oleksandr Kozlov <oleksandr@status.im>
+contributors:
+- Oleksandr Kozlov <oleksandr@status.im>
+- Prem Chaitanya Prathi <prem@status.im>
+- Franck Royer <franck@status.im>
+---
+
+## Table of contents
+
+<!-- TOC -->
+  * [Table of contents](#table-of-contents)
+  * [Abstract](#abstract)
+  * [Motivation](#motivation)
+  * [Syntax](#syntax)
+  * [API design](#api-design)
+    * [IDL](#idl)
+    * [Primitive types and general guidelines](#primitive-types-and-general-guidelines)
+    * [Language mappings](#language-mappings)
+    * [Application](#application)
+  * [The Waku API](#the-waku-api)
+    * [Initialise Waku node](#initialise-waku-node)
+      * [Type definitions](#type-definitions)
+      * [Function definitions](#function-definitions)
+      * [Predefined values](#predefined-values)
+      * [Extended definitions](#extended-definitions)
+  * [The Validation API](#the-validation-api)
+  * [Security/Privacy Considerations](#securityprivacy-considerations)
+  * [Copyright](#copyright)
+<!-- TOC -->
+
+## Abstract
+
+This document specifies an Application Programming Interface (API) that is RECOMMENDED for developers of the [WAKU2](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md) clients to implement,
+and for consumers to use as a single entry point to its functionalities.
+
+This API defines the RECOMMENDED interface for leveraging Waku protocols to send and receive messages. 
+Application developers SHOULD use it to access capabilities for peer discovery, message routing, and peer-to-peer reliability.
+
+## Motivation
+
+The accessibility of Waku protocols is capped by the accessibility of their implementations, and hence API.
+This RFC enables a concerted effort to draft an API that is simple and accessible, and provides an opinion on sane defaults.
+
+The API defined in this document is an opinionated-by-purpose method to use the more agnostic [WAKU2]() protocols.
+
+## Syntax
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## API design
+
+### IDL
+
+A custom Interface Definition Language (IDL) in YAML is used to define the Waku API.
+Existing IDL Such as OpenAPI, AsyncAPI or WIT do not exactly fit the requirements for this API.
+Hence, instead of having the reader learn a new IDL, we propose to use a simple IDL with self-describing syntax.
+
+An alternative would be to choose a programming language. However, such choice may express unintended opinions on the API.
+
+### Primitive types and general guidelines
+
+- No `default` means that the value is mandatory, meaning a `default` value implies an optional parameter.
+- Primitive types are `string`, `int`, `bool`, `enum` and `uint`
+- Complex pre-defined types are:
+  - `struct`: object and other nested types.
```

Comment:
nit: by being biased by high level languages I would prefer `object` instead of `struct`